### ISSUE TITLE
Vipsthumbnail PDF resize issue when using dpi as option and preventing upscaling

### ISSUE THREAD

**Issue by le0daniel** [libvips/libvips#1824]
**Describe the bug**
When using `vipsthumbnail` to resize a pdf with the **DPI option** and preventing upscaling using `>` as following: 
```vipsthumbnail input.pdf[dpi=150] --size 2400x\> -o converted.png[strip]```
The upscaling uses the width and height of the PDF without consideration of the DPI (so using the default dpi of 72). This results in images with much lower quality and the wrong dimensions.

**To Reproduce**

1. Run following command with and without `dpi=150`. Results in same dimensions of (595  ×  842).
`vipsthumbnail input.pdf[dpi=150] --size 2400x\> -o converted.png[strip]`
2. Run the command with and without `--dpi=150`. The dimensions are (1240 x  1754) with dpi option and (595  ×  842) without dpi option
`vips pdfload input.pdf --dpi=150 converted.png`

**Expected behavior**
Respect the DPI given by the load options when preventing upscaling

**Actual behavior**
Uses the default PDF DPI of 72 to prevent upscaling.

**Screenshots**
Example **with** DPI=150 using vipsthumbnail and `>` to prevent upscaling. Image dimensions (595  ×  842)
![converted](https://user-images.githubusercontent.com/11374766/93686286-8bd15d80-fab5-11ea-9228-5a03ccd710c6.png)

Example **without** DPI=150 using vipsthumbnail and `>` to prevent upscaling. Image dimensions (595  ×  842)
![converted-2](https://user-images.githubusercontent.com/11374766/93686287-8c69f400-fab5-11ea-9acf-317ef83975f7.png)

Sample when using pdfload to convert to png with DPI = 150, Results in dimensions of (1240 x  1754)
![sample1](https://user-images.githubusercontent.com/11374766/93686324-cb984500-fab5-11ea-9975-d2e6c2b0afc6.png)

Sample when using pdfload to convert to png with default DPI = 72, Results in dimensions of (595 × 842)
![sample2](https://user-images.githubusercontent.com/11374766/93686311-b58a8480-fab5-11ea-9974-c2a73cca4472.png)

**Environment**
 - OS: MacOS 10.15.5
 - Vips: 8.10.1

**Comment by jcupitt** [comment]
Hi @le0daniel,

You're right, `[dpi=..]` can break pdf scaling. You can see this happening if you use the `--vips-info` flag to enable debug output:

```
$ vipsthumbnail nipguide.pdf[dpi=150] --size 2400x --vips-info
VIPS-INFO: 09:07:30.723: thumbnailing nipguide.pdf[dpi=150]
VIPS-INFO: 09:07:30.727: selected loader is VipsForeignLoadPdfFile
VIPS-INFO: 09:07:30.727: input size is 1240 x 1754
VIPS-INFO: 09:07:30.727: loading with factor 0.516667 pre-shrink
VIPS-INFO: 09:07:30.730: pre-shrunk size is 1152 x 1629
VIPS-INFO: 09:07:30.730: converting to processing space srgb
VIPS-INFO: 09:07:30.730: premultiplying alpha
VIPS-INFO: 09:07:30.730: residual scale 2.08333 x 2.08333
VIPS-INFO: 09:07:30.731: unpremultiplying alpha
VIPS-INFO: 09:07:30.731: thumbnailing nipguide.pdf[dpi=150] as ./tn_nipguide.jpg
```

You can see that it first gets the unscaled resolution at 1240 pixels across, then tries to scale that up by 1.0 / 0.52 to get 2400 pixels across. But pdfload just sees the scale setting and ignores DPI in this case, so it fetches an image of the wrong size.

I've patched pdfload to support setting both scale and DPI and I now see:

```
$ vipsthumbnail nipguide.pdf[dpi=150] --size 2400x --vips-info
VIPS-INFO: 09:34:41.879: thumbnailing nipguide.pdf[dpi=150]
VIPS-INFO: 09:34:41.883: selected loader is VipsForeignLoadPdfFile
VIPS-INFO: 09:34:41.883: input size is 1240 x 1754
VIPS-INFO: 09:34:41.883: loading with factor 0.516667 pre-shrink
VIPS-INFO: 09:34:41.885: pre-shrunk size is 2400 x 3395
VIPS-INFO: 09:34:41.885: converting to processing space srgb
VIPS-INFO: 09:34:41.885: premultiplying alpha
VIPS-INFO: 09:34:41.885: unpremultiplying alpha
VIPS-INFO: 09:34:41.886: thumbnailing nipguide.pdf[dpi=150] as ./tn_nipguide.jpg
```

This fix will be in 8.10.2. Thanks for the report!

**Comment by le0daniel** [comment]
Amazing, thank you very much @jcupitt 😃
